File: statement-loader/model/account.py


class Account:
    def __init__(self, id: int, account_type_id: int = None, name: str = None):
        self.id: int  = id
        self.account_type_id: int  = account_type_id
        self.name: str = name

    def __str__(self):
        return self.__dict__.__str__()

File: account-service/src/main/java/com/finance/manager/account_service/dto/Account.java
package com.finance.manager.account_service.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;

import java.math.BigDecimal;

@Builder
public record Account(
        @JsonProperty("account_id")
        int accountId,

        @JsonProperty("account_name")
        String accountName,

        @JsonProperty("statement_file_extensions")
        String statementFileExtensions,

        @JsonProperty("icon")
        String icon,

        @JsonProperty("account_type_1")
        String accountType1,

        @JsonProperty("account_type_2")
        String accountType2,

        @JsonProperty("account_type_3")
        String accountType3,
        
        @JsonProperty("account_type_id")
        int accountTypeId
) {

}